wpengine/faustjs
# Support Custom Post Type templates in FaustNX

## Description

This PR adds support for custom post type templates in the FaustNX template hierarchy, i.e.:

* `single-$posttype`
* `archive-$posttype`
* `single-$posttype-$slug`

## Testing

0. Have a CPT that is public and has archives enabled on your WordPress site
1. Generate possible types. This PR introduces a script in the example project to do this, but ideally this will be included in the FaustNX CLI after Telemetry is merged in. For now, follow the steps below to generate possible types:
2. Run `npm run build -w faust-nx` to build
3. Run `cd examples/next/faust-nx`
4. Run `NEXT_PUBLIC_WORDPRESS_URL=http://your-url.local node scripts/generatePossibleTypes.js`. This will update the `possibleTypes.json` file in the example project with your WordPress sites' possible types. Be sure to replace `http://your-url.local` with your WordPress sites' URL.
5. Run `cd ../../../`
6. Now, start the example project dev server, `npm run dev -w faust-nx-getting-started` from the monorepo root.
7. Go to the CPTs archive page in the headless frontend and confirm that the archive templates are available, i.e. `['archive-cars', 'archive']`
8. Go to a single CPT post in the headless frontend and confirm that the single templates are available, i.e. `[ 'single-cars-1779-delorion', 'single-cars', 'singular', 'index' ]`
